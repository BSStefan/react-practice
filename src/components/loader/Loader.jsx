// @flow
import * as React from 'react';
import LoaderComponent from 'react-loader-spinner';
import styled from 'styled-components';
import { brand } from '../../styles/colors';

const Loader = (props: {}) => (
  <StyledLoader>
    <LoaderComponent
      type="ThreeDots"
      color={brand}
      height="100"
      width="100"
    />
  </StyledLoader>
)

const StyledLoader = styled.div`
  height: 30vh;
  padding-top: 20%;
  text-align: center;
`

export default Loader;