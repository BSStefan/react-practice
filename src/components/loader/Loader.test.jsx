import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Loader from './Loader';

it('Should match snapshout', () => {
  const tree = renderer.create(<Loader />).toJSON();
  expect(tree).toMatchSnapshot();
})