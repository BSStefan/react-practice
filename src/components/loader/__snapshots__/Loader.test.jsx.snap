// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshout 1`] = `
<div
  className="sc-bdVaJa QHXwa"
>
  <div>
    <svg
      fill="rgba(0,0,0,.9)"
      height="100"
      viewBox="0 0 120 30"
      width="100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="15"
        cy="15"
        r="15"
      >
        <animate
          attributeName="r"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="15"
          repeatCount="indefinite"
          to="15"
          values="15;9;15"
        />
        <animate
          attributeName="fillOpacity"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="1"
          repeatCount="indefinite"
          to="1"
          values="1;.5;1"
        />
      </circle>
      <circle
        attributeName="fillOpacity"
        cx="60"
        cy="15"
        from="1"
        r="9"
        to="0.3"
      >
        <animate
          attributeName="r"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="9"
          repeatCount="indefinite"
          to="9"
          values="9;15;9"
        />
        <animate
          attributeName="fillOpacity"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="0.5"
          repeatCount="indefinite"
          to="0.5"
          values=".5;1;.5"
        />
      </circle>
      <circle
        cx="105"
        cy="15"
        r="15"
      >
        <animate
          attributeName="r"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="15"
          repeatCount="indefinite"
          to="15"
          values="15;9;15"
        />
        <animate
          attributeName="fillOpacity"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="1"
          repeatCount="indefinite"
          to="1"
          values="1;.5;1"
        />
      </circle>
    </svg>
  </div>
</div>
`;
